tools/telemetry-generator/test/appInsightsMemoryUsageTestHandler.test.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TelemetryClient } from "applicationinsights";
+const handler = require("../src/handlers/appInsightsMemoryUsageTestHandler");
+
+describe("appInsightsMemoryUsageTestHandler", () => {
+	let mockTelemetryClient: TelemetryClient;
+	let mockFileData: any;
+
+	beforeEach(() => {
+		mockTelemetryClient = {
+			trackMetric: jest.fn(),
+		} as unknown as TelemetryClient;
+
+		mockFileData = {
+			suiteName: "TestSuite",
+			benchmarks: [
+				{
+					benchmarkName: "Benchmark1",
+					customData: {
+						"Heap Used Avg": "123.45",
+						"Heap Used StdDev": "67.89",
+					},
+				},
+			],
+		};
+	});
+
+	it("should emit heap used avg and std dev metrics to Azure App Insights", () => {
+		handler(mockFileData, mockTelemetryClient);
+
+		expect(mockTelemetryClient.trackMetric).toHaveBeenCalledTimes(2);
+
+		expect(mockTelemetryClient.trackMetric).toHaveBeenCalledWith({
+			name: "TestSuite_Benchmark1_heapUsedAvg",
+			value: "123.45",
+			namespace: "performance_benchmark_memoryUsage",
+			properties: {
+				buildId: process.env.BUILD_ID,
+				branchName: process.env.BRANCH_NAME,
+				category: "performance",
+				eventName: "Benchmark",
+				benchmarkType: "MemoryUsage",
+				suiteName: "TestSuite",
+				testName: "Benchmark1",
+			},
+		});
+
+		expect(mockTelemetryClient.trackMetric).toHaveBeenCalledWith({
+			name: "TestSuite_Benchmark1_heapUsedStdDev",
+			value: "67.89",
+			namespace: "performance_benchmark_memoryUsage",
+			properties: {
+				buildId: process.env.BUILD_ID,
+				branchName: process.env.BRANCH_NAME,
+				category: "performance",
+				eventName: "Benchmark",
+				benchmarkType: "MemoryUsage",
+				suiteName: "TestSuite",
+				testName: "Benchmark1",
+			},
+		});
+	});
+
+	it("should skip metrics if values are not numbers", () => {
+		mockFileData.benchmarks[0].customData["Heap Used Avg"] = "NaN";
+		mockFileData.benchmarks[0].customData["Heap Used StdDev"] = "NaN";
+
+		handler(mockFileData, mockTelemetryClient);
+
+		expect(mockTelemetryClient.trackMetric).not.toHaveBeenCalled();
+	});
+
+	it("should log an error if emitting metric fails", () => {
+		const consoleErrorSpy = jest.spyOn(console, "error").mockImplementation(() => {});
+		mockTelemetryClient.trackMetric = jest.fn(() => {
+			throw new Error("Test error");
+		});
+
+		handler(mockFileData, mockTelemetryClient);
+
+		expect(consoleErrorSpy).toHaveBeenCalledWith(
+			expect.stringContaining("failed to emit metric"),
+			expect.any(Error),
+		);
+
+		consoleErrorSpy.mockRestore();
+	});
```

Comment:
Similar to another comment, I don't think in this case there's value in tests that check for what the handler logs in case of errors. I'd say remove this one.

Reply:
Removed